File: Daily challenge/Island Perimeter.py
class Solution:
    def islandPerimeter(self, grid: List[List[int]]) -> int:
        n = len(grid)
        m = len(grid[0])
        perimeter = 0
        for i in range(n):
            for j in range(m):
                if grid[i][j] == 1:
                    if (j > 0 and grid[i][j - 1] != 1) or j == 0:
                        perimeter += 1
                    if (i > 0 and grid[i - 1][j] != 1) or i == 0:
                        perimeter += 1
                    if (j < m - 1 and grid[i][j + 1] != 1) or j == m - 1:
                        perimeter += 1
                    if (i < n - 1 and grid[i + 1][j] != 1) or i == n - 1:
                        perimeter += 1
        return perimeter

File: Daily challenge/Island Perimeter.cpp
class Solution 
{
public:
    int islandPerimeter(vector<vector<int>>& grid) 
    {
        int n = grid.size();
        int m = grid[0].size();
        int perimeter = 0;
        for(int i = 0; i < n; i++)
        {
            for(int j = 0; j < m; j++)
            {
                if(grid[i][j] == 1)
                {
                    if((j > 0 && grid[i][j - 1] != 1) || j == 0) perimeter++;
                    if((i > 0 && grid[i - 1][j] != 1) || i == 0) perimeter++;
                    if((j < m-1 && grid[i][j + 1] != 1) || j == m - 1) perimeter++;
                    if((i < n-1 && grid[i + 1][j] != 1) || i == n - 1) perimeter++;
                }
            }
        }
        return perimeter;
    }
};